File: java/src/main/java/com/gaston/macbook/simplemvp/presentation/show_apod/ApodDetailContract.java
package com.gaston.macbook.simplemvp.presentation.show_apod;

import com.gaston.macbook.simplemvp.presentation.show_apod.model.Apod;

public interface ApodDetailContract {

    interface View {

        void showProgressBar();

        void hideProgressBar();

        void hideApod();

        void showApod();

        void showApodDetails(Apod apodData);

        void fetchApodDetails();

        void showDataFetchError();

        void reloadData();

        void expandApodImage();

        void showApodImageProgressBar();

        void hideApodImageProgressBar();
    }

    interface Presenter {

        void fetchApodData();
    }


}

File: kotlin/src/main/java/com/gaston/macbook/kotlinsimplemvp/presentation/show_apod/ApodDetailContract.kt
package com.gaston.macbook.kotlinsimplemvp.presentation.show_apod

import com.gaston.macbook.kotlinsimplemvp.presentation.show_apod.model.Apod

interface ApodDetailContract {

    interface View{

        fun showProgressBar()

        fun hideProgressBar()

        fun hideApod()

        fun showApod()

        fun showApodDetails(apodData: Apod)

        fun fetchApodDetails()

        fun showDataFetchError()

        fun reloadData()

        fun expandApodImage()

        fun showApodImageProgressBar()

        fun hideApodImageProgressBar()
    }

    interface Presenter {

        fun fetchApodData()
    }
}